article/_posts/2020-05-28-NGU.md

```diff
@@ -0,0 +1,102 @@
+
+The authors propose
+> $$(i)$$ defining an exploration bonus combining life-long and episodic novelty to learn exploratory strategies that can maintain exploration throughout the agent’s training process (to never give up), $$(ii)$$ to learn a family of policies that separate exploration and exploitation using a conditional architecture with shared weights
+
+## The never-give-up intrinsic reward
+
+![](/article/images/NGU/fig1.jpeg)
+
+The network is trained on the augmented reward $$r_t = r_{t}^{e} + \beta{r_{t}^{i}}$$. Figure 1 describes the intrinsic reward modules. The embedding network can be thought of a siamese network trained my maximizing $$p(a \mid x_t, x_{t+1})$$. When used to embed states, the classifier and $$x_{t+1}$$ are discarded. This allows the network to project the states into a space where as much useless information as possible has been discarded. The states are then saved in a buffer and clustered using k-NN so as to get a pseudo-count of the states. $$r_{t}^{episodic}$$ is defined by
+
+![](/article/images/NGU/2.jpeg)
+
+The article explains $$K$$ in detail, but it is a riff on the euclidian distance between the "controlled states".
+
+The life-long novelty estimator is built using Random Network Distillation[^1]. Basically, a convolutional network $$g$$ is trained to match the output of another randomly-initialized untrained CNN $$\hat{g}$$. The $$\alpha$$ parameter is a riff on the error $$\|\hat{g}(x_t;\theta) - g(x)\|^2$$ so that as $$\hat{g}$$ sees the episode more and more, its error decreases and so does $$\alpha_t$$.
+
+## The never-give-up agent
+
+Using intrinsic reward means changing the underlying MDP.
+> If the augmented reward $$r_t = r_{t}^{e} + {\beta}{r_{t}^{i}}$$ varies in ways unpredictable from the action and states, then the decision process may no longer be a MDP, but instead be a Partially Observed MDP (POMDP).
+
+To alleviate the problem, the authors $$(i)$$ use a recurrent agent based on R2D2[^2] and $$(ii)$$ also feed the intrinsic reward, amongst other things, as an input to the network.
+
+![](/article/images/NGU/fig17.jpeg)
+
+To train a variety of exploring and exploiting policies, the authors use a universal value function approximator (UVFA)[^3] $$Q(x, a, \beta)$$ to approximate the optimal value function w.r.t the reward of the form $$r_t = r_{t}^{e} + {\beta}{_i}{r_{t}^{i}}$$ where $$\beta{_i}$$ is apart of a family of $$N$$ values $$\{\beta{_i}\}_{i=0}^{N-1}$$ including $$\beta_{0} = 0$$. Having a large N means having a greater variety in behavior and therefore more efficient training.
+
+## Experiments
+
+The authors tested on the Atari suited as well as their own new environment: <i>Random Disco Maze</i>. 
+
+![](/article/images/NGU/fig2.jpeg)
+
+The agent, in green, must navigate the maze, in black. If the agent takes an action that would make it go outside the maze, the experiment stops. At each time step, the colors of the background changes, and there is no extrinsic reward.
```

Comment:
_the colors of the background change_
or
_the color of the background changes_